### PR TITLE
[misc] Disable the offline cache if printing ir is enabled

### LINKED ISSUE #6218: ti.init(print_ir=True) fails to print IR when kernel is cached
**Describe the bug**
```python
import taichi as ti

ti.init(print_ir=True)

@ti.kernel
def foo():
    pass

foo()
```

Run this script twice and only the first time it prints IR of the `foo` kernel.

**Additional comments**
When a user sets `print_ir=True`, he probably wants the IR to be printed out **consistently** across multiple runs.

### PR DESCRIPTION
**Pull request by PGZXB**
Issue: fixes #6218

### CODE REVIEW

File: taichi/util/offline_cache.cpp [hunk 1]
```diff
@@ -5,6 +5,19 @@ namespace taichi::lang::offline_cache {
 constexpr std::size_t offline_cache_key_length = 65;
 constexpr std::size_t min_mangled_name_length = offline_cache_key_length + 2;
 
+void disable_offline_cache_if_needed(CompileConfig *config) {
+  TI_ASSERT(config);
+  if (config->offline_cache) {
+    if (config->print_preprocessed_ir || config->print_ir ||
+        config->print_accessor_ir) {
```
Comment (strongoier) [inline taichi/util/offline_cache.cpp:12]:
```suggestion
        config->print_accessor_ir || config->print_evaluator_ir) {
```

File: taichi/util/offline_cache.cpp [hunk 2]
```diff
@@ -5,6 +5,19 @@ namespace taichi::lang::offline_cache {
...
+      config->offline_cache = false;
+      TI_WARN(
+          "Disable offline_cache because print_preprocessed_ir or print_ir or "
+          "print_accessor_ir is enabled");
```
Comment (strongoier) [inline taichi/util/offline_cache.cpp:16]:
```suggestion
          "print_accessor_ir or print_evaluator_ir is enabled");
```

**Review by strongoier — APPROVED** [review]
LGTM!

File: taichi/util/offline_cache.cpp (hunk 2 above)
Comment (PGZXB) [reply taichi/util/offline_cache.cpp:16]: Thanks. But the evaluators won't be cached, so I think `print_evaluator_ir` will always work correctly. WDYT?

(see:
1. LLVM

https://github.com/taichi-dev/taichi/blob/master/taichi/codegen/codegen.cpp#L103-L104

3. GFX

https://github.com/taichi-dev/taichi/blob/master/taichi/cache/gfx/cache_manager.cpp#L151-L155

5. Metal (in #6227)
https://github.com/taichi-dev/taichi/blob/e559866e933b13d507bf8308d3da3c88e42fc03a/taichi/cache/metal/cache_manager.cpp#L28-L30

Comment (strongoier) [reply taichi/util/offline_cache.cpp:16]: Thanks for pointing this out. I agree that `print_evaluator_ir` will work properly.

**Pull request merged**